packages/dev/docs/src/client.js

```diff
@@ -183,8 +186,95 @@ function Hamburger() {
   );
 }
 
+function DocSearch() {
+  const client = algoliasearch('1V1Q59JVTR', '44a7e2e7508ff185f25ac64c0a675f98');
+  const searchIndex = client.initIndex('react-spectrum');
+  const searchOptions = {
+    highlightPreTag: `<mark class="${docsStyle.docSearchBoxMark}">`,
+    highlightPostTag: '</mark>'
+  };
+
+  const [searchValue, setSearchValue] = useState('');
+  const [predictions, setPredictions] = useState(null);
+  const [suggestions, setSuggestions] = useState(null);
+
+  let updatePredictions = ({hits}) => {
+    setPredictions(hits);
+    let sections = [];
+    hits.forEach(prediction => {
+      let hierarchy = prediction.hierarchy;
+      let objectID = prediction.objectID;
+      let lvl0 = hierarchy.lvl0;
+      let section = sections.find(section => section.title === lvl0);
+      if (!section) {
+        section = {title: lvl0, items: []};
+        sections.push(section);
+      }
+      let text = [];
+      let textValue = [];
+      for (let i = 1; i < 6; i++) {
+        if (hierarchy[`lvl${i}`]) {
+          text.push(prediction._highlightResult.hierarchy[`lvl${i}`].value);
+          textValue.push(hierarchy[`lvl${i}`]);
+        }
+      }
+      section.items.push(
+        <Item key={objectID} textValue={textValue.join(' | ')}>
+          <Text><span dangerouslySetInnerHTML={{__html: DOMPurify.sanitize(text.join(' | '))}} /></Text>
+          {
+            prediction.content &&
```

Comment:
Should this be `prediction._snippetResult.content`? I don't see a description displayed, and some of the results seem to be duplicated. Maybe because they are different parts of the same page that are meant to be highlighted?

<img width="307" alt="Screen Shot 2022-08-30 at 4 37 42 PM" src="https://user-images.githubusercontent.com/19409/187562232-b37c24d0-041a-46ef-9670-c1f42b6b0b86.png">

Reply:
See 42e258bf524db9531272964579e9564c22a8ff47

Reply:
I'm still not seeing a description. I see above you check for `prediction.content` but below use `prediction._snippetResult.content`. Should these match?

Reply:
`prediction.content` is the unformatted content string, for hits without a description `prediction.content` is `null` and `prediction._snippetResult.content` will not exist. Some hits will not include a description, like for example when the query is returned as part of a heading at any level. When the hit is returned from body content, it will include `prediction.content` and `prediction._snippetResult.content.value` will contain the content formatted as an html string, including the `mark` tag that highlights the query in context.

Reply:
Example of JSON returned for "DateRangePicker" `H1` hit, which has `"content": null`:
```json
    {
      "version": "",
      "tags": [],
      "url": "https://react-spectrum.adobe.com/react-spectrum/DateRangePicker.html#daterangepicker",
      "url_without_variables": "https://react-spectrum.adobe.com/react-spectrum/DateRangePicker.html#daterangepicker",
      "url_without_anchor": "https://react-spectrum.adobe.com/react-spectrum/DateRangePicker.html",
      "anchor": "daterangepicker",
      "content": null,
      "content_camel": null,
      "lang": "en-US",
      "language": "en-US",
      "type": "lvl1",
      "no_variables": false,
      "weight": {
        "pageRank": "0",
        "level": 90,
        "position": 0
      },
      "hierarchy": {
        "lvl0": "Documentation",
        "lvl1": "DateRangePicker",
        "lvl2": null,
        "lvl3": null,
        "lvl4": null,
        "lvl5": null,
        "lvl6": null
      },
      "recordVersion": "v2",
      "hierarchy_radio": {
        "lvl0": null,
        "lvl1": "DateRangePicker",
        "lvl2": null,
        "lvl3": null,
        "lvl4": null,
        "lvl5": null,
        "lvl6": null
      },
      "hierarchy_camel": [
        {
          "lvl0": "Documentation",
          "lvl1": "DateRangePicker",
          "lvl2": null,
          "lvl3": null,
          "lvl4": null,
          "lvl5": null,
          "lvl6": null
        }
      ],
      "hierarchy_radio_camel": {
        "lvl0": null,
        "lvl1": "DateRangePicker",
        "lvl2": null,
        "lvl3": null,
        "lvl4": null,
        "lvl5": null,
        "lvl6": null
      },
      "objectID": "0-https://react-spectrum.adobe.com/react-spectrum/DateRangePicker.html",
      "_snippetResult": {
        "version": {
          "value": "",
          "matchLevel": "none"
        },
        "url": {
          "value": "https://react-spectrum.adobe.com/react-spectrum/<mark class="algolia-docsearch-suggestion--highlight">Date</mark>RangePicker.html#<mark class="algolia-docsearch-suggestion--highlight">date</mark>rangepicker",
          "matchLevel": "full"
        },
        "url_without_variables": {
          "value": "https://react-spectrum.adobe.com/react-spectrum/<mark class="algolia-docsearch-suggestion--highlight">Date</mark>RangePicker.html#<mark class="algolia-docsearch-suggestion--highlight">date</mark>rangepicker",
          "matchLevel": "full"
        },
        "url_without_anchor": {
          "value": "https://react-spectrum.adobe.com/react-spectrum/<mark class="algolia-docsearch-suggestion--highlight">Date</mark>RangePicker.html",
          "matchLevel": "full"
        },
        "anchor": {
          "value": "<mark class="algolia-docsearch-suggestion--highlight">date</mark>rangepicker",
          "matchLevel": "full"
        },
        "lang": {
          "value": "en-US",
          "matchLevel": "none"
        },
        "language": {
          "value": "en-US",
          "matchLevel": "none"
        },
        "type": {
          "value": "lvl1",
          "matchLevel": "none"
        },
        "weight": {
          "pageRank": {
            "value": "0",
            "matchLevel": "none"
          },
          "level": {
            "value": "90",
            "matchLevel": "none"
          },
          "position": {
            "value": "0",
            "matchLevel": "none"
          }
        },
        "hierarchy": {
          "lvl0": {
            "value": "Documentation",
            "matchLevel": "none"
          },
          "lvl1": {
            "value": "<mark class="algolia-docsearch-suggestion--highlight">Date</mark>RangePicker",
            "matchLevel": "full"
          }
        },
        "recordVersion": {
          "value": "v2",
          "matchLevel": "none"
        },
        "hierarchy_radio": {
          "lvl1": {
            "value": "<mark class="algolia-docsearch-suggestion--highlight">Date</mark>RangePicker",
            "matchLevel": "full"
          }
        },
        "hierarchy_camel": [
          {
            "lvl0": {
              "value": "Documentation",
              "matchLevel": "none"
            },
            "lvl1": {
              "value": "<mark class="algolia-docsearch-suggestion--highlight">Date</mark>RangePicker",
              "matchLevel": "full"
            }
          }
        ],
        "hierarchy_radio_camel": {
          "lvl1": {
            "value": "<mark class="algolia-docsearch-suggestion--highlight">Date</mark>RangePicker",
            "matchLevel": "full"
          }
        }
      },
      "_highlightResult": {
        "hierarchy": {
          "lvl0": {
            "value": "Documentation",
            "matchLevel": "none",
            "matchedWords": []
          },
          "lvl1": {
            "value": "<mark class="algolia-docsearch-suggestion--highlight">Date</mark>RangePicker",
            "matchLevel": "full",
            "fullyHighlighted": false,
            "matchedWords": [
              "date"
            ]
          }
        },
        "hierarchy_camel": [
          {
            "lvl0": {
              "value": "Documentation",
              "matchLevel": "none",
              "matchedWords": []
            },
            "lvl1": {
              "value": "<mark class="algolia-docsearch-suggestion--highlight">Date</mark>RangePicker",
              "matchLevel": "full",
              "fullyHighlighted": false,
              "matchedWords": [
                "date"
              ]
            }
          }
        ]
      },
      "_rankingInfo": {
        "nbTypos": 0,
        "firstMatchedWord": 4000,
        "proximityDistance": 0,
        "userScore": 6336,
        "geoDistance": 0,
        "geoPrecision": 1,
        "nbExactWords": 0,
        "words": 1,
        "filters": 0
      }
    }
```

Reply:
And here is the example JSON for "useTimeField > Anatomy", which includes `content`:

```json
    {
      "version": "",
      "tags": [],
      "url": "https://react-spectrum.adobe.com/react-aria/useTimeField.html#anatomy",
      "url_without_variables": "https://react-spectrum.adobe.com/react-aria/useTimeField.html#anatomy",
      "url_without_anchor": "https://react-spectrum.adobe.com/react-aria/useTimeField.html",
      "anchor": "anatomy",
      "content": "useDateSegment returns props for an individual time segment:",
      "content_camel": "useDateSegment returns props for an individual time segment:",
      "lang": "en-US",
      "language": "en-US",
      "type": "content",
      "no_variables": false,
      "weight": {
        "pageRank": "0",
        "level": 0,
        "position": 15
      },
      "hierarchy": {
        "lvl0": "Documentation",
        "lvl1": "useTimeField",
        "lvl2": "Anatomy",
        "lvl3": null,
        "lvl4": null,
        "lvl5": null,
        "lvl6": null
      },
      "recordVersion": "v2",
      "hierarchy_radio": {
        "lvl0": null,
        "lvl1": null,
        "lvl2": null,
        "lvl3": null,
        "lvl4": null,
        "lvl5": null,
        "lvl6": null
      },
      "hierarchy_camel": [
        {
          "lvl0": "Documentation",
          "lvl1": "useTimeField",
          "lvl2": "Anatomy",
          "lvl3": null,
          "lvl4": null,
          "lvl5": null,
          "lvl6": null
        }
      ],
      "hierarchy_radio_camel": {
        "lvl0": null,
        "lvl1": null,
        "lvl2": null,
        "lvl3": null,
        "lvl4": null,
        "lvl5": null,
        "lvl6": null
      },
      "objectID": "15-https://react-spectrum.adobe.com/react-aria/useTimeField.html",
      "_snippetResult": {
        "version": {
          "value": "",
          "matchLevel": "none"
        },
        "url": {
          "value": "https://react-spectrum.adobe.com/react-aria/useTimeField.html#anatomy",
          "matchLevel": "none"
        },
        "url_without_variables": {
          "value": "https://react-spectrum.adobe.com/react-aria/useTimeField.html#anatomy",
          "matchLevel": "none"
        },
        "url_without_anchor": {
          "value": "https://react-spectrum.adobe.com/react-aria/useTimeField.html",
          "matchLevel": "none"
        },
        "anchor": {
          "value": "anatomy",
          "matchLevel": "none"
        },
        "content": {
          "value": "use<mark class="algolia-docsearch-suggestion--highlight">Date</mark>Segment returns props for an individual time segment:",
          "matchLevel": "full"
        },
        "content_camel": {
          "value": "useDateSegment returns props for an individual time segment:",
          "matchLevel": "none"
        },
        "lang": {
          "value": "en-US",
          "matchLevel": "none"
        },
        "language": {
          "value": "en-US",
          "matchLevel": "none"
        },
        "type": {
          "value": "content",
          "matchLevel": "none"
        },
        "weight": {
          "pageRank": {
            "value": "0",
            "matchLevel": "none"
          },
          "level": {
            "value": "0",
            "matchLevel": "none"
          },
          "position": {
            "value": "15",
            "matchLevel": "none"
          }
        },
        "hierarchy": {
          "lvl0": {
            "value": "Documentation",
            "matchLevel": "none"
          },
          "lvl1": {
            "value": "useTimeField",
            "matchLevel": "none"
          },
          "lvl2": {
            "value": "Anatomy",
            "matchLevel": "none"
          }
        },
        "recordVersion": {
          "value": "v2",
          "matchLevel": "none"
        },
        "hierarchy_camel": [
          {
            "lvl0": {
              "value": "Documentation",
              "matchLevel": "none"
            },
            "lvl1": {
              "value": "useTimeField",
              "matchLevel": "none"
            },
            "lvl2": {
              "value": "Anatomy",
              "matchLevel": "none"
            }
          }
        ]
      },
      "_highlightResult": {
        "content": {
          "value": "use<mark class="algolia-docsearch-suggestion--highlight">Date</mark>Segment returns props for an individual time segment:",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "date"
          ]
        },
        "hierarchy": {
          "lvl0": {
            "value": "Documentation",
            "matchLevel": "none",
            "matchedWords": []
          },
          "lvl1": {
            "value": "useTimeField",
            "matchLevel": "none",
            "matchedWords": []
          },
          "lvl2": {
            "value": "Anatomy",
            "matchLevel": "none",
            "matchedWords": []
          }
        },
        "hierarchy_camel": [
          {
            "lvl0": {
              "value": "Documentation",
              "matchLevel": "none",
              "matchedWords": []
            },
            "lvl1": {
              "value": "useTimeField",
              "matchLevel": "none",
              "matchedWords": []
            },
            "lvl2": {
              "value": "Anatomy",
              "matchLevel": "none",
              "matchedWords": []
            }
          }
        ]
      },
      "_rankingInfo": {
        "nbTypos": 0,
        "firstMatchedWord": 28001,
        "proximityDistance": 0,
        "userScore": 2293,
        "geoDistance": 0,
        "geoPrecision": 1,
        "nbExactWords": 0,
        "words": 1,
        "filters": 0
      }
    }
```

Reply:
If you open https://reactspectrum.blob.core.windows.net/reactspectrum/138527ee8b97ab7a7c26afe042ccd6cc8a2dd1f7/docs/index.html and search for a term that is less commonly used in headings, like for example "correct", DocSearch will return more results that include `content`.

Reply:
oh ok. I still haven't managed to find one but I believe you. Got no results for "correct".

Reply:
> oh ok. I still haven't managed to find one but I believe you. Got no results for "correct".

Press down arrow to expand the menu and you should see results with "correct" highlighted in the description. I'm not exactly sure why the results hide after entering longer strings, like "correct".